src/get_env.c

```diff
@@ -55,6 +55,27 @@ rcutils_get_env(const char * env_name, const char ** env_value)
   return NULL;
 }
 
+const char *
+rcutils_get_home_dir(void)
+{
+  const char * homedir;
+
+  if (rcutils_get_env("HOME", &homedir) == NULL && *homedir != '\0') {
+    // The HOME environment variable was set and is non-empty, return it.
+    return homedir;
+  }
+
+  // OK, we didn't find a HOME variable, try USERPROFILE (mostly for
+  // Windows)
```

Comment:
Isn't this only set on Windows? Then this could be wrapped in a `ifdef _WIN32` block.

Reply:
Yeah, I went back and forth a bit on that logic.  It really only applies to Windows (where the most common case I've seen is that HOME is not set and USERPROFILE is set).  On the other hand, it doesn't really hurt on Unix.  But I don't feel strongly about it, so I can wrap it.

Reply:
Fixed in 3352d19